### PR TITLE
Remove wrapper fee unshielding

### PR DESCRIPTION
**Pull request by grarco**
## Describe your changes

Partially addresses #2597.

Removes the optional fee unshielding `Transaction` object from `WrapperTx`. Removes/refactors associated functions, types and tests. Removes fee unshielding support from the client and the sdk.

**NOTE**: with this PR only transactions that pay fees from the signer balance are accepted. If you need to test masp transactions you'll need to have enough funds in a transparent balance.

## Indicate on which release or other PRs this topic is based on

#3103 (diffs for review: https://github.com/anoma/namada/pull/3217/files/7f4b6c1e6656ec44194562ad1e167b09d24170f4..cf52c605e33279f4f464b210a8e215a7138be809)

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

**Comment by grarco** [comment]
@yito88 this pr changes the structure of the `Tx` once more so we'll probably need another update of Hermes

File: crates/apps/src/lib/node/ledger/shell/prepare_proposal.rs [hunk 1]
```diff
@@ -294,10 +293,10 @@ where
         super::replay_protection_checks(&tx, temp_state).map_err(|_| ())?;
 
         // Check fees and extract the gas limit of this transaction
+        // TODO: check if masp fee payment is required
```
Comment (cwgoes) [inline crates/apps/src/lib/node/ledger/shell/prepare_proposal.rs:296]: Can we start a practice of referencing a GitHub issue every time we write a TODO?

File: crates/light_sdk/src/transaction/transfer.rs [hunk 2]
```diff
@@ -21,7 +21,7 @@ impl Transfer {
         target: Address,
         token: Address,
         amount: DenominatedAmount,
-        // FIXME: handle masp here
+        // TODO: handle masp here
```
Comment (cwgoes) [inline crates/light_sdk/src/transaction/transfer.rs:24]: Ditto, all TODOs should reference an issue.

**Review by cwgoes — APPROVED** [review]
Changes ACK, we should track TODOs in issues

**Review by batconjurer — APPROVED** [review]
Apart from a complaint I have about the pr this is based on top of, this looks fine to me.

File: crates/apps/src/lib/cli/client.rs [hunk 3]
```diff
@@ -327,7 +327,6 @@ impl CliApi {
                             chain_ctx.shielded,
                             &client,
                             &io,
-                            args.batch_size,
```
[the remark this answers is not in the document]
Comment (grarco) [reply crates/apps/src/lib/cli/client.rs]: This should be fixed now in the base PR

**Pull request merged**